stat/distmat/permutation.go

```diff
@@ -0,0 +1,53 @@
+// Copyright ©2020 The Gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package distmat
+
+import (
+	"golang.org/x/exp/rand"
+
+	"gonum.org/v1/gonum/mat"
+)
+
+// UniformPermutation is a distribution over permutation matrices
+type UniformPermutation struct {
+	src rand.Source
+}
+
+// NewUniformPermutation constructs a new permutation matrix
+// generator using the given random source.
+func NewUniformPermutation(src rand.Source) *UniformPermutation {
+	up := UniformPermutation{src: src}
+	return &up
+}
+
+// Matrix draws a random permutuation matrix of dimension n.
+func (up *UniformPermutation) Matrix(n int) *mat.Dense {
```

Comment:
We have another matrix distribution in this package, `Wishart`, in which the function sampling a random matrix is called `RandSymTo`. Neither this name or `Matrix` seems very fitting to me. We should decide on a consistent API now that we have more than one distribution in the package.

@kortschak ?